src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -24,41 +24,47 @@ extern "C" {
 
 /** @brief The interface type */
 enum ddsrt_iftype {
-  DDSRT_IFTYPE_UNKNOWN,
-  DDSRT_IFTYPE_WIRED,
-  DDSRT_IFTYPE_WIFI
+  DDSRT_IFTYPE_UNKNOWN, ///< An unknown interface
+  DDSRT_IFTYPE_WIRED, ///< A wired interface (e.g. ethernet)
+  DDSRT_IFTYPE_WIFI ///< A wifi interface
 };
 
 /** @brief Linked list of interface addresses */
 struct ddsrt_ifaddrs {
-  struct ddsrt_ifaddrs *next;
-  char *name;
-  uint32_t index;
-  uint32_t flags;
-  enum ddsrt_iftype type;
-  struct sockaddr *addr;
-  struct sockaddr *netmask;
-  struct sockaddr *broadaddr;
+  struct ddsrt_ifaddrs *next; ///< next pointer of the linked list
+  char *name; ///< name of the interface
+  uint32_t index; ///< the index of the interface
+  uint32_t flags; ///< flags from 'struct ifaddrs'
```

Comment:
You do realize that `struct ifaddrs` doesn't exist on Windows and that this reference is quite opaque to anyone using that ... operating system? If you search for, e.g., `IFF_MULTICAST` you'll find it, but the problem with that is that it presupposes someone knows what to look for.

I think it would be better to write something like
```suggestion
  uint32_t flags; ///< interface flags, e.g., IFF_MULTICAST, IFF_LOOPBACK, IFF_BROADCAST, ...
```
that at least gives someone not familiar with these interfaces a good starting point

Reply:
Okay, I really just wasn't sure what to do with this. Essentially most of this unit isn't really our code, they are just very thin wrappers (with arguably a bit better error handling) around OS calls. Therefore I'm tempted to just say some form of "go look at the OS documentation".